[Controller] Set function status to `unhealthy` instead of `error` during fast-fail check (#3165)
nuclio/nuclio: docs/reference/function-configuration/function-configuration-reference.md

| internalInvocationUrls | []string | A list of internal urls to invoke the function                                                    |
| externalInvocationUrls | []string | A list of external urls to invoke the function, including ingresses and external-ip:function-port |

> ### Function state
> 
> Here is a list of all possible function states:
> 

Reviewer: ```suggestion
### Function state (`state`)

The `state` fields described the current function status, and can be one of the following:

```

Also - add the "Function State" link in the table of contents (sub-title to `status`)

| **State**                       | **Description**                                                                                                                                    |

> |:--------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------|

Reviewer: I would reorder then states a bit (happy, intermediate, bad):
- ready
- imported
- scaledToZero
- building
- waitingForResourceConfiguration
- waitingForScaleResourceFromZero
- waitingForScaleResourceToZero
- unhealthy
- error

| ready                           | Function is deployed successfully and ready to process events.                                                                                     |

> | imported                        | Function is imported from file and hasn't been deployed yet.                                                                                       |

Reviewer: ```suggestion
| imported                        | Function has been imported but hasn't been deployed yet.                                                                                       |
```

> | imported                        | Function has been imported but hasn't been deployed yet.                                                                                           |

Reviewer 2: | imported | Function is imported but not yet deployed. |

> | error                           | Error occurred during function deployment and cannot be rectified without redeployment.                                                            |

Reviewer: ```suggestion
| error                           | An error occurred during function deployment that cannot be rectified without redeployment.                                                            |
```

> | unhealthy                       | Error occurred during function deployment, which can be resolved over time. For example, issues such as insufficient resources or a missing image. |

Reviewer: ```suggestion
| unhealthy                       | An error occurred during function deployment, which can be resolved over time. For example, issues with insufficient resources or a missing image. |
```

| waitingForResourceConfiguration | Function waits for resources to be ready. For instance, in case of k8s function waits for deployment/pods and etc.                                 |
| waitingForScaleResourceFromZero | Function is scaling up from zero replicas.                                                                                                         |
| waitingForScaleResourceToZero   | Function is scaling down to zero replicas.                                                                                                         |
| error                           | An error occurred during function deployment that cannot be rectified without redeployment.                                                        |

> | unhealthy                       | An error occurred during function deployment, which can be resolved over time. For example, issues with insufficient resources or a missing image. |

Reviewer: ```suggestion
| unhealthy                       | An error occurred during function deployment, which might be resolved over time, and might require redeployment. For example, issues with insufficient resources or a missing image. |
```
